include/deal.II/base/tensor.h

```diff
@@ -1993,7 +1993,7 @@ DEAL_II_CONSTEXPR inline DEAL_II_ALWAYS_INLINE
   typename Tensor<rank_1 + rank_2 - 2,
                   dim,
                   typename ProductType<Number, OtherNumber>::type>::tensor_type
-    result;
```

Comment:
Relates to #8474. I do the same over there.

Reply:
Ah, sorry, I missed that. I'm happy to remove the first commit if we get further with that other PR more quickly.

Reply:
This one is probably easier to review and merge than #8474. :slightly_smiling_face: